src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -60,41 +95,61 @@ const createIframeClickHandler = (container, collect) => {
       target.tagName.toLowerCase() === "a" ? target : target.closest("a");
 
     if (anchor) {
-      if (ANCHOR_HREF_REGEX.test(anchor.href)) {
-        const matches = ANCHOR_HREF_REGEX.exec(anchor.href);
-
-        const action = matches.length >= 2 ? matches[1] : "";
-        const interaction = matches.length >= 3 ? matches[2] : "";
-
-        if (interaction === "clicked") {
-          const uuid = anchor.getAttribute("data-uuid");
-          // eslint-disable-next-line no-console
-          console.log(`clicked ${uuid}`);
-          // TODO: collect analytics
-          // collect({
-          //   eventType: INTERACT
-          // });
+      const parts = anchor.href.split("?");
+      const actionPart = parts[0].split("://")[1];
+      let action = "";
+      let interaction = "";
+      let link = "";
+      if (parts.length > 1) {
+        const queryParams = new URLSearchParams(parts[1]);
+        action = actionPart;
+        interaction = queryParams.get("interaction") || "";
+        link = queryParams.get("link") || "";
+        const uuid = anchor.getAttribute("data-uuid") || "";
+        // eslint-disable-next-line no-console
+        console.log(`clicked ${uuid}`);
+        // TODO: collect analytics
+        // collect({
+        //   eventType: INTERACT,
+        //   eventSource: "inAppMessage",
+        //   eventData: {
+        //     action,
+        //     interaction
+        //   }
+        // });
+        if (link && interaction === "clicked") {
+          link = decodeURIComponent(link);
+          setWindowLocationHref(link);
+        } else if (action === "dismiss") {
+          const containerElement = document.getElementById(ELEMENT_TAG_ID);
```

Comment:
I propose you extract the dismiss code into a separate function called `dismissMessage` and call it here.

the code could be simplified like so...

```js
import { removeElements } from "../utils";

const dismissMessage = () =>
  [OVERLAY_TAG_CLASSNAME, ELEMENT_TAG_CLASSNAME].forEach(removeElements);
```